include/seqan3/argument_parser/detail/format_ctd.hpp

```diff
@@ -0,0 +1,484 @@
+                argument_advanced = "false";
+            }
+            else if (spec == ADVANCED)
+            {
+                argument_required = "false";
+                argument_advanced = "true";
+            }
+            else
+            {
+                argument_required = "false";
+                argument_advanced = "false";
+            }
+            
+            // Register CTD 'ITEM' node creation callback.
+            append_ctd_item_node(argument_name,
+                                 get_type_as_gkn_string(value,
+                                                        validator),
+                                 desc,
+                                 "",
+                                 "*.*",
+                                 argument_required,
+                                 argument_advanced,
+                                 "");
+        });
+    }
+
+    /*!\brief Adds calls for appending 'clielement' and 'ITEM' nodes to the CTD DOM tree to be evaluated at parse time.
+     *
+     * \param[in]  value    The variable in which to store the given command line argument.
+     * \param[in]  short_id The short identifier for the flag (e.g. 'i').
+     * \param[in]  long_id  The long identifier for the flag (e.g. "integer").
+     * \param[in]  desc     The description of the flag.
+     * \param[in]  spec     Advanced flag specification. see seqan3::option_spec.
+     *
+     * \details
+     * For the sake of the CTD exporter, add_flag is the same of add_option, with a constrained option_type and
+     * a default validator.
+     */
+    void add_flag(bool & value,
+                  char const short_id,
+                  std::string const & long_id,
+                  std::string const & desc,
+                  option_spec const & spec) 
+    {
+        add_option(value, 
+                   short_id,
+                   long_id,
+                   desc,
+                   spec,
+                   default_validator<bool> {});
+    }
+
+    /*!\brief Adds calls for appending 'clielement' and 'ITEM' nodes to the CTD DOM tree to be evaluated at parse time.
+     *
+     * \tparam option_type    The type of variable in which to store the given command line argument.
+     * \tparam validator_type The type of validator applied to the value after parsing.
+     *
+     * \param[out] value     The variable in which to store the given command line argument.
+     * \param[in]  desc      The description of the positional option.
+     * \param[in]  validator The validator applied to the value after parsing (callable).
+     */
+    template<typename option_type, typename validator_type>
+    void add_positional_option(option_type & value,
+                               std::string const & desc,
+                               validator_type && validator) 
+    {
+        unsigned argument_id = args_counter;
+
+        clielement_argument_callbacks.push_back([this,
+						 argument_id] (std::string app_name) {
```

Comment:
the formatting looks odd

Reply:
Yes, that's definitely a kind of typo I missed :) I will change it !